src/Microsoft.OpenApi.OData.Reader/PathItem/PathItemHandler.cs

```diff
@@ -101,5 +113,54 @@ protected virtual void AddOperation(OpenApiPathItem item, OperationType operatio
             IOperationHandler operationHander = provider.GetHandler(Path.Kind, operationType);
             item.AddOperation(operationType, operationHander.CreateOperation(Context, Path));
         }
+
+        /// <summary>
+        /// Set the parameters information for <see cref="OpenApiPathItem"/>
+        /// </summary>
+        /// <param name="item">The <see cref="OpenApiPathItem"/>.</param>
+        protected virtual void SetParameters(OpenApiPathItem item)
+        {
+            foreach (ODataKeySegment keySegment in Path.OfType<ODataKeySegment>())
+            {
+                IDictionary<string, string> mapping = ParameterMappings[keySegment];
+                foreach (var parameter in Context.CreateKeyParameters(keySegment, mapping))
+                {
+                    AppendParameter(item, parameter);
+                }
+            }
+
+            // Add the route prefix parameter v1{data}
+            if (Context.Settings.RoutePathPrefixProvider != null && Context.Settings.RoutePathPrefixProvider.Parameters != null)
+            {
+                foreach (var parameter in Context.Settings.RoutePathPrefixProvider.Parameters)
+                {
+                    item.Parameters.Add(parameter);
```

Comment:
don't you have an add range method that'd prevent manually looping here?

Reply:
@baywet `Parameters` are declared with a type of `IList<T>` and not `List<T>`. `AddRange` is defined in `List<T>`, so for us to use it we need to convert the IList to a List e.g. `item.Parameters.ToList().AddRange(...)`. The conversion uses the same `foreach` loop internally, so we don't get any performance gain. The resulting code is more concise though.

Reply:
ToList() is noop if the backing type is in effect a List if I remember things correctly, you'd have to reassign the properly after doing this though, your call.